crates/shared/src/trade_finding.rs

```diff
@@ -142,11 +142,16 @@ impl From<PriceEstimationError> for TradeError {
     fn from(err: PriceEstimationError) -> Self {
         match err {
             PriceEstimationError::NoLiquidity => Self::NoLiquidity,
-            PriceEstimationError::UnsupportedOrderType => Self::UnsupportedOrderType,
-            PriceEstimationError::DeadlineExceeded => Self::DeadlineExceeded,
+            PriceEstimationError::UnsupportedOrderType(order_type) => {
+                Self::UnsupportedOrderType(order_type)
+            }
+            PriceEstimationError::UnsupportedToken { token, .. } => {
+                Self::UnsupportedOrderType(format!("{token:#x}"))
+            }
+            PriceEstimationError::ZeroAmount => Self::UnsupportedOrderType("zero amount".into()),
```

Comment:
These error conversions seem surprising.
I assumed `UnsupportedOrderType` exclusively means that `buy` or `sell` orders are not supported by that estimator. Was that changed to include anything about an order not being supported (weird amounts, order type, hooks, signatures, send ETH to SC, ...)?
If so maybe something like `UnsupportedOrderConfiguration` might be more fitting?

Reply:
Right, technically for those types of errors we have the `PartialValidationError` type. Technically all queries that reach a price estimator should already be checked for that (ideally the type on `Query.in_amount` would be NonZeroU256).

PriceEstimationError::ZeroAmount is currently checked for in the Baseline solver and in the sanitizing estimator. I don't have a full picture of the control flow to say with confidence that we can never receive these requests there (and could potentially panic if removed)